File: single-language-analysis/dataset-files/java/TestSmell071.java
// Original URL: https://github.com/INFO-ESIPE/TP-Programmation_Concurrente/blob/48457a69d6f38d906dda5d47965b1e91bbcb3194/TP%204/src/VoteTest.java#L53-L67

public class TestSmell071 {
    @Test
    public void ManyVotes() throws Exception {
        var vote = new Vote(2);
        for(var i = 0; i < 4; i++) {
            Thread.ofPlatform().start(() -> {
                try {
                    Thread.sleep(1_000);
                    assertEquals("0", vote.vote("1"));
                } catch (InterruptedException e) {
                    throw new AssertionError(e);
                }
            });
        }
        assertEquals("0", vote.vote("0"));
    }
}

File: single-language-analysis/dataset-files/python/test_smell_071.py
import pytest

class TestVoteTest:
    def test_smell_071(self):
        vote = Vote(2)
        
        def thread_function():
            try:
                time.sleep(1)
                assert vote.vote("1") == "0"
            except InterruptedException as e:
                raise AssertionError(e)

        threads = []
        for i in range(4):
            thread = threading.Thread(target=thread_function)
            threads.append(thread)
            thread.start()

        for thread in threads:
            thread.join()

        assert vote.vote("0") == "0"